### DISCUSSION TITLE
Move the Add Parameter in extensions edition

### DISCUSSION THREAD

**Discussion by UlisesFreitas** [4ian/GDevelop#4397] [Enhancement ideas (and their technical discussions)]
## Description
Hi mates I wonder if some of this changes can be implemented in the future versions of GDevelop
Is just to make the Add Paramter button floating or add a new button in the topbar, something like this.
Check this too https://forum.gdevelop.io/t/move-the-add-parameter-in-extensions-edition/42646

## Solution suggested
![addparameter](https://user-images.githubusercontent.com/2497411/195996957-6469c776-f9b3-49ec-84d1-0aacccc2ae15.jpeg)

[tristanrhodes](https://forum.gdevelop.io/u/tristanrhodes)
I love these ideas!

Lock the “Add parameter” button to the bottom so it doesn’t move when we scroll. This is something that we have improved in other places in GDevelop, such as the list of animations of a sprite.

Adding an option to “Add a new parameter…” from the three-dot drop-down menu of parameters. This would enable inserting parameters in the right place. GDevelop already offers this type of functionality in several areas, such as “Add a new object…”

Add the ability to drag-and-drop parameters to reorder them. I think a drag handle could be added to the left side of the text “Parameter #1”. I think this would be very useful!

Any actions that change the re-ordering of parameters must also edit the numbering in the extension sentence so creators don’t have to manually update it (this has bitten several of us in the past; we forgot to edit the sentence after moving parameters)

I support each of these ideas, but I’m not sure how easy/hard they are to implement nor the priority they will be assigned. [@UlisesFreitas](https://forum.gdevelop.io/u/ulisesfreitas) it might be a good idea to submit a new GitHub issue and link it to this forum post.

**Comment by Bouh** [comment]
A UI/UX designer is working on the new user interface, they are currently working on other parts of the software that will set the foundation for the other parts of the editor including the extensions interface.

No improvements to the extension interface are planned until the new UI is delivered.
Nevertheless, the ideas forge the imagination and the feedbacks are important to have a more solid and pleasant interface.
I turn this _issue_ into a _discussion_.

**Comment by tristanbob** [comment]
@Bouh thanks for the update.  I want to add that the first 3 ideas I proposed for "function parameters" should also be improved in "behavior properties".
